src/components/Router/index.js

```diff
@@ -42,10 +42,18 @@ class CrossDeviceMobileRouter extends Component {
 
   setConfig = (actions) => (data) => {
     const {token, steps, documentType, step} = data
-    this.setState({token, steps, step})
+    const crossDeviceClientSteps = this.replaceLastStep(steps)
+    this.setState({token, steps: crossDeviceClientSteps, step})
     actions.setDocumentType(documentType)
   }
 
+  replaceLastStep = (steps) => {
```

Comment:
I don't think this logic should happen here - why can't it happen in `StepComponentMap`?